flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/BatchTableEnvironment.scala

```diff
@@ -170,7 +171,7 @@ abstract class BatchTableEnvironment(
   private[flink] def explain(table: Table, extended: Boolean): String = {
     val ast = table.getRelNode
     val optimizedPlan = optimize(ast)
-    val dataSet = translate[Row](optimizedPlan) (TypeExtractor.createTypeInfo(classOf[Row]))
+    val dataSet = translate[Row](optimizedPlan, ast.getRowType) (new RowTypeInfo())
```

Comment:
This is the problem described in the issue description of FLINK-5662. The optimize does not change the types of the fields but their names. This might be a Calcite bug but until then we need the logical type for the sink conversion.